### ISSUE TITLE
Improved MD038 fixing for words stuck to backtick

### ISSUE THREAD

**Issue by nschonni** [DavidAnson/markdownlint#481]
Currently the fix for MD038 correctly trims the interior of the spans, but sometimes that interior space is because of a missing space after the backtick.
EX:
```md
My `code `should be good
```
Gets fixed to:
```md
My `code`should be good
```
But ideally it would detect a non-whitespace character trailing the lead/closing backtick and add the padding like:
```md
My `code` should be good
```
